packages/core-flows/src/product/workflows/create-products.ts

```diff
@@ -10,6 +17,77 @@ export const createProductsWorkflow = createWorkflow(
   (
     input: WorkflowData<WorkflowInput>
   ): WorkflowData<ProductTypes.ProductDTO[]> => {
-    return createProductsStep(input.products)
+    // Passing prices to the product module will fail, we want to keep them for after the product is created.
+    const productWithoutPrices = transform({ input }, (data) =>
+      data.input.products.map((p) => ({
+        ...p,
+        variants: p.variants?.map((v) => ({
+          ...v,
+          prices: undefined,
+        })),
+      }))
+    )
+
+    const createdProducts = createProductsStep(productWithoutPrices)
+
+    // Note: We rely on the same order of input and output when creating products here, make sure that assumption holds
+    const variantsWithAssociatedPrices = transform(
+      { input, createdProducts },
+      (data) => {
+        return data.createdProducts
+          .map((p, i) => {
+            const inputProduct = data.input.products[i]
+            return p.variants?.map((v, j) => ({
+              id: v.id,
+              prices: (inputProduct?.variants?.[j] as any).prices,
+            }))
+          })
+          .flat()
+          .filter((v) => v.prices?.length > 0)
+      }
+    )
+
+    const createdPriceSets = createPriceSetsStep(variantsWithAssociatedPrices)
+
+    const variantAndPriceSets = transform(
+      { variantsWithAssociatedPrices, createdPriceSets },
+      (data) =>
+        data.variantsWithAssociatedPrices.map((variant, i) => ({
+          variant: variant,
+          price_set: data.createdPriceSets[i],
+        }))
+    )
+
+    const variantAndPriceSetLinks = transform(
+      { variantAndPriceSets },
+      (data) => {
+        return {
+          links: data.variantAndPriceSets.map((entry) => ({
+            variant_id: entry.variant.id,
+            price_set_id: entry.price_set.id,
+          })),
+        }
+      }
+    )
+
+    createVariantPricingLinkStep(variantAndPriceSetLinks)
+
+    return transform(
+      {
+        createdProducts,
+        variantAndPriceSets,
+      },
+      (data) => {
+        return data.createdProducts.map((product) => ({
+          ...product,
+          variants: product.variants?.map((variant) => ({
+            ...variant,
+            price_set: data.variantAndPriceSets.find(
```

Comment:
suggestion: we could also just refetch in the create API using the remote query and skip this dance.